### PR TITLE
[SEDB-51] convert user redux actions to use thunk

### PR DESCRIPTION
**Pull request by tsangela**
- thunk
- mock pending state
- loaders to indicate pending state
- reducer looks big and redundant, but i've already done as much as i could to reduce the file size and reducers are naturally pretty huge
- fix modal width
- delete unused `StyleeProfileView`
- fix empty birthdate format

note: removed welcome message on login since it doesn't play well with thunk conversion, can figure out how to add back later

user-thunk

### CODE REVIEW

File: client/src/app/Views/SignupView.js [hunk 1]
```diff
@@ -1,39 +1,49 @@
...
 import { RedirectFooter, CenterWrapper } from '../../components/globals/index';
 import StyleeAccountCreationComponent from '../../components/AccountCreation/StyleeAccountCreationComponent';
 import { routes } from '../../resources/routes';
-import { signUp } from '../../redux/actions/userActions';
 import userTypes from '../../resources/userTypes';
 import StylerAccountCreationComponent from '../../components/AccountCreation/StylerAccountCreationComponent';
 
+const formName = 'normal_signup';
+const headerText = 'Create an account.';
+const submitTitle = 'Sign Up';
```
Comment (Amym123) [inline client/src/app/Views/SignupView.js:14]: should we have these in a constant file later?

File: client/src/redux/actions/userActions.js [hunk 2]
```diff
@@ -1,5 +1,62 @@
 import { userActionTypes } from './types';
 
+export const signUp = (username, password, userType) => {
```
Comment (jameskanCa) [inline client/src/redux/actions/userActions.js:3]: These look to be an action creator file, perhaps we should stay consistent with the naming pattern that's been used? Name this userActionCreators?

File: client/src/components/DetailedView/DetailedPostView.js [hunk 3]
```diff
@@ -60,7 +60,7 @@ class DetailedPostView extends React.Component {
     return (
       <div>
         <Modal
-          width={1300}
+          width="95vw"
```
Comment (Amym123) [inline client/src/components/DetailedView/DetailedPostView.js:63]: just wondering, will this also change the size of components inside the modal? how does it look if we shrink it

File: client/src/components/PostCreation/SuggestionPostCreationComponent.js [hunk 4]
```diff
@@ -111,32 +133,36 @@ const StylerSuggestionCreationComponent = () => {
   const [form] = Form.useForm();
 
   const onFinish = (values) => {
-    console.log('Received values of form:', values); // todo(atsang): delete
```
Comment (tsangela) [inline client/src/components/PostCreation/SuggestionPostCreationComponent.js:114]: only deleted this -- everything else is linter

File: client/src/components/DetailedView/DetailedPostView.js (hunk 3 above)
Comment (tsangela) [reply client/src/components/DetailedView/DetailedPostView.js:63]: doesn't shrink contents! but changing to `"95vw"` gets rid of overflow issue if your screen is smaller than that, but still displays the modal to be as wide as looks nice. hope to do that in the responsive ticket (kinda out of scope for this one)

File: client/src/app/Views/SignupView.js (hunk 1 above)
Comment (tsangela) [reply client/src/app/Views/SignupView.js:14]: hm but we pass these as props??

Comment (tsangela) [reply client/src/app/Views/SignupView.js:14]: i would say i wouldn't put them in a constant file, unless we put all other text (e.g. 'Save', 'Edit Profile') in a constant file

File: client/src/components/PostCreation/SuggestionPostCreationComponent.js (hunk 4 above)
Comment (tsangela) [reply client/src/components/PostCreation/SuggestionPostCreationComponent.js:114]: nvm i added it back forgot about lint

File: client/src/redux/reducers/userReducer.js [hunk 5]
```diff
@@ -2,13 +2,7 @@ import moment from 'moment';
...
 import userTypes from '../../resources/userTypes';
 
-const initialState = {
-  isLoggedIn: false,
-  userType: null,
-  profile: null,
-};
-
-const mockInitUserInfo = {
+const MOCK_PROFILE = {
```
Comment (tsangela) [inline client/src/redux/reducers/userReducer.js:5]: moved this up here so we can easily put it in `initialState` to quickly bypass login page during testing (must be declared before)

File: client/src/components/DetailedView/DetailedPostView.js (hunk 3 above)
Comment (tsangela) [reply client/src/components/DetailedView/DetailedPostView.js:63]: i guess i should have done this change in the other ticket too 😓 got too excited

**Comment by jameskanCa** [comment]
LGTM!

**Pull request merged**